service-bus.api.md

```diff
@@ -0,0 +1,337 @@
+
+// @public
+export interface ReceiveBatchOptions extends OperationOptions, WaitTimeOptions {}
+
+// @public
+export interface ReceivedMessage extends ServiceBusMessage {
+  readonly _amqpMessage: AmqpMessage;
+  readonly deadLetterSource?: string;
+  readonly deliveryCount?: number;
+  readonly enqueuedSequenceNumber?: number;
+  readonly enqueuedTimeUtc?: Date;
+  readonly expiresAtUtc?: Date;
+  lockedUntilUtc?: Date;
+  readonly lockToken?: string;
+  readonly sequenceNumber?: Long;
+}
+
+// @public
+export interface ReceivedMessageWithLock extends ReceivedMessage {
+  abandon(propertiesToModify?: { [key: string]: any }): Promise<void>;
+  complete(): Promise<void>;
+  deadLetter(
+    options?: DeadLetterOptions & {
+      [key: string]: any;
+    }
+  ): Promise<void>;
+  defer(propertiesToModify?: { [key: string]: any }): Promise<void>;
+  renewLock(): Promise<Date>;
+}
+
+// @public
+export interface Receiver<ReceivedMessageT> {
+  close(): Promise<void>;
+  diagnostics: {
+    peek(maxMessageCount?: number): Promise<ReceivedMessage[]>;
+    peekBySequenceNumber(
+      fromSequenceNumber: Long,
+      maxMessageCount?: number
+    ): Promise<ReceivedMessage[]>;
+  };
+  entityPath: string;
+  getMessageIterator(options?: GetMessageIteratorOptions): AsyncIterableIterator<ReceivedMessageT>;
+  isReceivingMessages(): boolean;
+  receiveBatch(maxMessages: number, options?: ReceiveBatchOptions): Promise<ReceivedMessageT[]>;
+  receiveDeferredMessage(
+    sequenceNumber: Long,
+    options?: OperationOptions
+  ): Promise<ReceivedMessageT | undefined>;
+  receiveDeferredMessages(
+    sequenceNumbers: Long[],
+    options?: OperationOptions
+  ): Promise<ReceivedMessageT[]>;
+  receiveMode: "peekLock" | "receiveAndDelete";
+  subscribe(handlers: MessageHandlers<ReceivedMessageT>, options?: SubscribeOptions): void;
+}
+
+export { RetryOptions };
+
+// @public
+export interface RuleDescription {
+  action?: string;
+  filter?: string | CorrelationFilter;
+  name: string;
+}
+
+// @public
+export interface Sender {
+  cancelScheduledMessage(sequenceNumber: Long): Promise<void>;
+  cancelScheduledMessages(sequenceNumbers: Long[]): Promise<void>;
+  close(): Promise<void>;
+  createBatch(options?: CreateBatchOptions): Promise<ServiceBusMessageBatch>;
+  isClosed: boolean;
+  scheduleMessage(scheduledEnqueueTimeUtc: Date, message: ServiceBusMessage): Promise<Long>;
+  scheduleMessages(scheduledEnqueueTimeUtc: Date, messages: ServiceBusMessage[]): Promise<Long[]>;
+  send(message: ServiceBusMessage): Promise<void>;
+  sendBatch(messageBatch: ServiceBusMessageBatch): Promise<void>;
+}
+
+// @public
+export class ServiceBusClient {
+  constructor(connectionString: string, options?: ServiceBusClientOptions);
+  constructor(
+    hostName: string,
+    tokenCredential: TokenCredential,
+    options?: ServiceBusClientOptions
+  );
+  close(): Promise<void>;
+  getDeadLetterReceiver(
+    queueName: string,
+    receiveMode: "peekLock",
```

Comment:
We should create an issue to discuss the need to make receive mode a mandatory parameter